tools/perf/scripts/bench_run_log.py

```diff
@@ -276,13 +279,32 @@ def single_run(self):
             logger.error(f"Failed to generate payload: {e}")
             return
 
+        metrics_count_per_step = round(self.num_metrics * self.percentage // 100)
+        logger.info(f"metrics_count_per_step {metrics_count_per_step}")
+
+        subset_payloads = []
+        # Because we are logging only a "subset" of the metrics in each step
+        # we will prepare the subset-payload for each step ahead of time
+        if metrics_count_per_step != self.num_metrics:
+            for s in range(self.num_steps):
+                keys = list(payloads[0].keys())
+                start_index = (s * metrics_count_per_step) % self.num_metrics
+                end_index = start_index + metrics_count_per_step
+                subset_payloads.append(
+                    {key: payloads[0][key] for key in keys[start_index:end_index]}
+                )
```

Comment:
do you always want the same metric keys on each round? or the goal is just to have a specific percentage, because an alternative implementation if we only care about the count is to randomize the indices in the relevant amount.
For example if you have 10 keys and your percentage is 0.5.
on each iteration you can randomize. something like: `random.sample(payloads[0], metrics_count_per_step)`